databricks_cli/configure/config.py

```diff
@@ -27,55 +27,34 @@
 import six
 
 from databricks_cli.configure.provider import DEFAULT_SECTION, get_config_for_profile
-from databricks_cli.utils import error_and_quit
-from databricks_cli.sdk import ApiClient, DbfsService, WorkspaceService, JobsService, \
-    ClusterService, ManagedLibraryService
+from databricks_cli.utils import error_and_quit, InvalidConfigurationError
+from databricks_cli.sdk import ApiClient
 
 
-def require_config(function):
+def provide_api_client(function):
+    """
+    All callbacks wrapped by require_config expect the argument ``profile`` to be passed in.
+    """
     @six.wraps(function)
     def decorator(*args, **kwargs):
-        config = get_config_for_profile(DEFAULT_SECTION)
+        profile = kwargs.pop('profile')
+        config = get_config_for_profile(profile)
         if not config.is_valid:
-            error_and_quit(('You haven\'t configured the CLI yet! '
-                            'Please configure by entering `{} configure`').format(sys.argv[0]))
+            raise InvalidConfigurationError(profile)
+        kwargs['api_client'] = _get_api_client(config)
+
         return function(*args, **kwargs)
     decorator.__doc__ = function.__doc__
     return decorator
 
 
 def profile_option(f):
-    return click.option('--profile', required=False, default=DEFAULT_SECTION)(f)
+    return click.option('--profile', required=False, default=DEFAULT_SECTION,
+                        help='CLI connection profile to use.')(f)
```

Comment:
Mention whether this is case sensitive or not. Btw, we should think more about case sensitivity. Convert all letters to uppercase might result different meanings in some languages. And in the current implementation, we expect users use all uppercase letters if they modify the `~/.databrickscfg` file directly and add sections there. Say if I add a section call `[test]` it won't be pick up by `--profile test`, which is definitely not the expected behavior.

Reply:
Btw, `databricks fs ls --profile XXX` works but not `databricks --profile XXX fs ls` nor `databricks fs --profile XXX ls`. This is different from AWS CLI experience, which users are more familiar with. We can address this in a follow-up PR.